pipsi.py

```diff
@@ -404,13 +405,17 @@ def uninstall(repo, package, yes):
 @click.pass_obj
 def list_cmd(repo):
     """Lists all scripts installed through pipsi."""
-    click.echo('Packages and scripts installed through pipsi:')
-    for venv, scripts in repo.list_everything():
-        if not scripts:
-            continue
-        click.echo('  Package "%s":' % venv)
-        for script in scripts:
-            click.echo('    ' + script)
+    list_of_non_empty_venv = [(venv, scripts)
+                              for venv, scripts in repo.list_everything()
+                              if scripts]
+    if list_of_non_empty_venv:
+        click.echo('Packages and scripts installed through pipsi:')
+        for venv, scripts in list_of_non_empty_venv:
+            click.echo('  Package "%s":' % venv)
+            for script in scripts:
+                click.echo('    ' + script)
+    else:
```

Comment:
Fair enough, I also overlooked something that lead me to believe we could eliminate the list comprehension.